src/parser/parser.mly

```diff
@@ -295,11 +302,19 @@ top_handler_cases:
 
 (* XXX allow patterns here *)
 top_handler_case:
-  | op=var_name xs=top_patt_maybe_var* y=top_handler_checking DARROW t=term           { (op, (xs, y, t)) }
-  | op=PREFIXOP x=top_patt_maybe_var y=top_handler_checking DARROW t=term
-    { (fst op, ([x], y, t)) }
-  | x1=top_patt_maybe_var op=infix x2=top_patt_maybe_var y=top_handler_checking DARROW t=term
-    { (fst op, ([x1;x2], y, t)) }
+  | op=var_name xs=top_patt_maybe_var* y=top_handler_checking DARROW t=term
+    { (op, (xs, y, t)) }
+  | oploc=prefix x=top_patt_maybe_var y=top_handler_checking DARROW t=term
+    { let (op, _) = oploc in
+      (op, ([x], y, t))
+    }
+  | x1=top_patt_maybe_var oploc=infix x2=top_patt_maybe_var y=top_handler_checking DARROW t=term
+    { let (op, _) = oploc in
+      (op, ([x1;x2], y, t)) }
+
+top_patt_maybe_var:
+  | x=patt_var                   { x }
```

Comment:
It would be more consistent to use an option type here to indicate that `_` was used.